package/MDAnalysis/coordinates/GRO.py

```diff
@@ -223,15 +216,20 @@ def _read_first_frame(self):
         if len(unitcell) == 3:
             # special case: a b c --> (a 0 0) (b 0 0) (c 0 0)
             # see Timestep.dimensions() above for format (!)
-            self.ts._unitcell[:3] = unitcell
+            self.ts.dimensions = np.r_[unitcell, [90., 90., 90.]]
         elif len(unitcell) == 9:
-            self.ts._unitcell[:] = unitcell  # fill all
+            self.ts.dimensions = _gmx_to_dimensions(unitcell)
         else:  # or maybe raise an error for wrong format??
             warnings.warn("GRO unitcell has neither 3 nor 9 entries --- might be wrong.")
```

Comment:
I know this is historical, but shouldn't we just throw an error if this happens? "not sure about this" doesn't fill one with much confidence.